classes/ActionScheduler_ActionFactory.php

```diff
@@ -4,6 +4,31 @@
  * Class ActionScheduler_ActionFactory
  */
 class ActionScheduler_ActionFactory {
+
+	/**
+	 * @param int $action_id The action's ID in the data store
+	 * @param string $hook The hook to trigger when this action runs
+	 * @param string $status The action's status in the data store
+	 * @param string $claim_id The identifier for the claim this action belongs to, if any, derived from ActionScheduler_ActionClaim::get_id()
+	 * @param array $args Args to pass to callbacks when the hook is triggered
+	 * @param ActionScheduler_Schedule $schedule The action's schedule
+	 * @param string $group A group to put the action in
+	 *
+	 * @return string The ID of the stored action
+	 */
+	public function get_stored_action_instance( $action_id, $hook, $status, $claim_id = '', array $args = array(), ActionScheduler_Schedule $schedule = NULL, $group = '' ) {
+
+		if ( ActionScheduler_Store::STATUS_PENDING === $status ) {
```

Comment:
Seeing this here makes me think that perhaps it would be better to have the various status constants in an interface, rather than an abstract class. But I recognize also that we may not really want to attempt to change that right now, so consider this a public note rather than a request for a change.

With that said, would it make sense here to use a `switch` statement instead of if/elseif/else? Or perhaps to add a filter to the final class string for extensibility? Those are real questions, not necessarily change requests.

Reply:
A `switch` makes sense because it reduces the changeset if we introduce additional status based classes in future. I think a filter makes sense also to allow 3rd parties to add their own classes too. I've changed both of those and rebased in the the changes.